lib/clientv2/watch_e2e_test.go

```
@@ -0,0 +1,182 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	"github.com/projectcalico/libcalico-go/lib/testutils"
+	"github.com/projectcalico/libcalico-go/lib/watch"
+)
+
+// Perform additional watch tests
+var _ = testutils.E2eDatastoreDescribe("Additional watch tests", testutils.DatastoreEtcdV3, func(config apiconfig.CalicoAPIConfig) {
+
+	ctx := context.Background()
+	numEvents := 10000
+	numWatchers := 1000
+
+	Describe("Test prefix deletion of the datastore", func() {
+		It("should receive watch events for each of the deleted keys", func() {
+			c, err := New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
+
+			By("Creating 10000 resources")
+			deleteEvents := []watch.Event{}
+			for ii := 1; ii <= numEvents; ii++ {
+				name := fmt.Sprintf("peer-%08d", ii)
+				peer := &apiv2.BGPPeer{
+					ObjectMeta: metav1.ObjectMeta{Name: name},
+					Spec: apiv2.BGPPeerSpec{
+						PeerIP:   "1.2.3.4",
+						ASNumber: numorstring.ASNumber(ii),
+					},
+				}
+				_, outError := c.BGPPeers().Create(ctx, peer, options.SetOptions{})
+				Expect(outError).NotTo(HaveOccurred())
+				deleteEvents = append(deleteEvents, watch.Event{
+					Type:     watch.Deleted,
+					Previous: peer,
+				})
+			}
+
+			By("Listing all the resources")
+			outList, outError := c.BGPPeers().List(ctx, options.ListOptions{})
+			Expect(outError).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(numEvents))
+
+			By("Creating a watcher, watching the current resource version")
+			w, _ := c.BGPPeers().Watch(ctx, options.ListOptions{ResourceVersion: outList.ResourceVersion})
```

why are we ignoring this error? we should assert the error here